p2p/host/autorelay/autorelay.go

```diff
@@ -246,8 +360,29 @@ func (ar *AutoRelay) connect(ctx context.Context, pi peer.AddrInfo) bool {
 		return false
 	}
 
-	// tag the connection as very important
-	ar.host.ConnManager().TagPeer(pi.ID, "relay", 42)
+	// wait for identify to complete in at least one conn so that we can check the supported protocols
+	conns := ar.host.Network().ConnsToPeer(pi.ID)
+	if len(conns) == 0 {
+		return false
+	}
+
+	ready := make(chan struct{}, len(conns))
+	for _, conn := range conns {
+		go func(conn network.Conn) {
+			select {
+			case <-ar.host.IDService().IdentifyWait(conn):
+				ready <- struct{}{}
+			case <-ctx.Done():
+			}
+		}(conn)
+	}
+
+	select {
+	case <-ready:
```

Comment:
This will leave all but the first go routine blocked on their `IdentifyWait`, doesn't it? Maybe this would a good use case for https://pkg.go.dev/golang.org/x/sync/errgroup#WithContext?

Reply:
No, the channel is buffered for `len(conns)`, so we should be fine.